lib/jsonb_accessor/macro.rb

```diff
@@ -65,7 +65,15 @@ def jsonb_accessor(jsonb_attribute, field_types)
           names_and_store_keys.each do |name, store_key|
             define_method("#{name}=") do |value|
               super(value)
-              new_values = (public_send(jsonb_attribute) || {}).merge(store_key => public_send(name))
+
+              attribute_value = public_send(name)
+              # Rails always saves time based on `default_timezone`. Since #as_json considers timezone, manual conversion is needed
+              if attribute_value.acts_like?(:time)
+                default_timezone = ActiveRecord::Base.default_timezone
+                attribute_value = (default_timezone == :utc ? attribute_value.utc : attribute_value.in_time_zone).strftime("%FT%R:%S.%LZ")
```

Comment:
I think `.strftime("%FT%R:%S.%LZ")` is equivalent to `.iso8601`.

Reply:
Not quite. `#iso8601` has timezone information, this formatter "discards" TZ
```
irb(main):031:0> Time.zone.now.strftime("%FT%R:%S.%LZ")
=> "2021-05-27T09:14:23.876Z"
irb(main):032:0> Time.zone.now.iso8601
=> "2021-05-27T09:14:35-03:00"
```